### PR TITLE
[Statistics Feature] - Personal Statistics

### LINKED ISSUE #46: [Statistics Feature] - Cost statistics
### Description

As a user, I want to be able to view cost statistics so that I know how much I'm spending over periods of time

### Acceptance Criteria

- [ ] Cost time periods (monthly, yearly, all time)
- [ ] Different chart views
- [ ] Rspec tests
- [ ] Cucumber tests

### PR DESCRIPTION
**Pull request by coraenglish**
### Pivotal Tracker
- https://www.pivotaltracker.com/story/show/181144548

### Description / PR Commit Message

- Added personal statistics page
- Styled page
- Applied styling to Sitewide page

image

### Guidance for Review

- Added rspec tests as features since the statistics page is mainly focused on viewing static content from subscription stats

### Author's Checklist
- [x] Wrote tests?
- [x] Linked to issue in Github?
- [x] Tests passed?
- [x] Test deployment to gatekeeper-tamu-test?

### CODE REVIEW

File: app/views/statistics/index.html.erb [hunk 1]
```diff
@@ -1,40 +1,116 @@
...
+                  <th scope="col">Created On</th>
+                </tr>
+              </thead>
+              <tbody>
+              <% current_user.subscriptions.order(:cost_per_month).each do |sub| %>
+                  <tr>
+                    <td>$<%= sub.cost_per_month %></td>
+                    <td>$<%= ((((Time.now - sub.created_at) / 60 / 60 / 24).round / 30 + 1) * sub.cost_per_month) %></td>
+                    <td><%= sub.subscription_name %></td>
+                    <td><a href="https://<%= sub.url %>" target=_blank><%= sub.url %></a></td>
```
Comment (rebeccamcfadden) [inline app/views/statistics/index.html.erb]: use regex to fix this. that way if the user enters a url that has the https:// it doesn't break. i'm actually including a fix for this if you just want to remove the additional string
```suggestion
                    <td><a href="<%= sub.url %>" target=_blank><%= sub.url %></a></td>

```

**Review by rebeccamcfadden — CHANGES_REQUESTED** [review]
Left one request for changes but other than that LGTM. Could use some cucumber tests.

The graphs look great!

File: app/views/statistics/index.html.erb (hunk 1 above)
Comment (coraenglish) [reply app/views/statistics/index.html.erb]: Gotcha, that makes sense! Just changed it

**Review by rebeccamcfadden — APPROVED** [review]

**Comment by rebeccamcfadden** [comment]
Good to merge!

**Pull request merged**